src/ParallelTempering.jl

```diff
@@ -126,13 +180,121 @@ function gather_print(replica::Replica, data, fname::String)
     return nothing
 end
 
+"""
+Print xyz formatted (Lx, Ly, Lz, Sx, Sy, Sz) configurations to file 
+"""
+function xyz_to_file(sys::SpinSystem{3}, output::IOStream)
+    sites = reinterpret(reshape, Float64, collect(sys.lattice))
+    spins = reinterpret(reshape, Float64, collect(sys.sites))
+
+    xyz = hcat(
+        hcat([vec(sites[i,:,:,:,:]) for i in 1:3]...),
+        hcat([vec(spins[i,:,:,:,:]) for i in 1:3]...)
+    )
+
+    for r in eachrow(xyz)
+        @printf(output, "%f\t%f\t%f\t%f\t%f\t%f\n", r...)
+    end
+    println(output, "")
+
+    return nothing
+end
```

Comment:
I find the series of `hcat`'s a bit hard to follow, is this version fine with you? (I don't think the `collect`'s are necessary?)

```julia
function xyz_to_file(sys::SpinSystem{3}, output::IOStream)
    sites = reinterpret(reshape, Float64, sys.lattice)
    spins = reinterpret(reshape, Float64, sys.sites)
    xyz = vcat(sites, spins)
    xyz = reshape(xyz, 6, :)

    for c in eachcol(xyz)
        @printf(output, "%f\t%f\t%f\t%f\t%f\t%f\n", c...)
    end
    println(output, "")

    return nothing
end
```

Reply:
This is much better, thanks.